Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.hxx

```diff
@@ -0,0 +1,188 @@
+}
+
+/** Compute Curvature */
+template <typename TInputMesh>
+void
+TriangleMeshCurvatureCalculator<TInputMesh>::Compute()
+{
+  MeshConstPointer inputMesh = this->m_TriangleMesh;
+  if (inputMesh == nullptr)
+  {
+    itkExceptionMacro("First set the Input Triangle Mesh to perform computation");
+  }
+
+  itkDebugMacro(<< "Executing Curvature");
+
+  if (this->GetCurvatureType() == Self::CurvatureGauss)
+  {
+    this->GetGaussCurvature(inputMesh);
+  }
+  else
+  {
+    itkExceptionMacro("Only Gauss Curvature type available");
+    return;
+  }
+}
+
+//--------------------------------------------
+template <typename TInputMesh>
+void
+TriangleMeshCurvatureCalculator<TInputMesh>::GetGaussCurvature(MeshConstPointer inputMesh)
+{
+  if (inputMesh->GetNumberOfCells())
+  {
+    this->ComputeGaussCurvature(inputMesh);
+  }
+
+  itkDebugMacro("Gauss Curvature Computation : Done");
+}
+
+
+template <typename TInputMesh>
+void
+TriangleMeshCurvatureCalculator<TInputMesh>::ComputeGaussCurvature(MeshConstPointer inputMesh)
+{
+  MeshPointType e0, e1, e2;
+  double        A, alpha0, alpha1, alpha2;
+
+  // other data
+  const unsigned int Nv = inputMesh->GetNumberOfPoints();
+
+  const std::unique_ptr<double[]> K(new double[Nv]);
+  const std::unique_ptr<double[]> dA(new double[Nv]);
+  double                          pi2 = 2.0 * itk::Math::pi;
+  for (unsigned int k = 0; k < Nv; k++)
+  {
+    K[k] = pi2;
+    dA[k] = 0.0;
+  }
+
+  CellsContainerConstPointer  outCells = inputMesh->GetCells();
+  CellsContainerConstIterator cellsItr = outCells->Begin();
+
+  while (cellsItr != outCells->End())
+  {
+    CellType *               cellPointer = cellsItr.Value();
+    MeshPointIdConstIterator point_ids = cellPointer->GetPointIds();
+
+    MeshPointType v0 = inputMesh->GetPoint(point_ids[0]);
+    MeshPointType v1 = inputMesh->GetPoint(point_ids[1]);
+    MeshPointType v2 = inputMesh->GetPoint(point_ids[2]);
+
+    // edges
+    e0[0] = v1[0];
+    e0[1] = v1[1];
+    e0[2] = v1[2];
+    e0[0] -= v0[0];
+    e0[1] -= v0[1];
+    e0[2] -= v0[2];
+
+    e1[0] = v2[0];
+    e1[1] = v2[1];
+    e1[2] = v2[2];
+    e1[0] -= v1[0];
+    e1[1] -= v1[1];
+    e1[2] -= v1[2];
+
+    e2[0] = v0[0];
+    e2[1] = v0[1];
+    e2[2] = v0[2];
+    e2[0] -= v2[0];
+    e2[1] -= v2[1];
+    e2[2] -= v2[2];
+
+    alpha0 = itk::Math::pi - angle(e1.GetVnlVector(), e2.GetVnlVector());
+    alpha1 = itk::Math::pi - angle(e2.GetVnlVector(), e0.GetVnlVector());
+    alpha2 = itk::Math::pi - angle(e0.GetVnlVector(), e1.GetVnlVector());
+
+    // surf. area
+    A = double(std::abs(vnl_cross_3d((v1 - v0).GetVnlVector(), (v2 - v0).GetVnlVector()).two_norm() / 2.0));
+
+    dA[point_ids[0]] += A;
+    dA[point_ids[1]] += A;
+    dA[point_ids[2]] += A;
+    K[point_ids[0]] -= alpha1;
+    K[point_ids[1]] -= alpha2;
+    K[point_ids[2]] -= alpha0;
+
+    ++cellsItr;
+  }
+
+  // Allocate Memory to store the curvature output
+  this->m_GaussCurvatureData = MeshPointDataContainer::New();
```

Comment:
This should have been done before the while loop. And its values set inside the loop. That way, there would have been no need to allocate `K` and `dA` arrays.

Reply:
K and dA are needed as they accumulate results for points in the loop.